### PR TITLE
refactor(bazel): dont rely on language target to downlevel for loop

### PR DESCRIPTION
**Pull request by Toxicable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[x] Refactoring (no functional changes, no api changes)
```

## What is the current behavior?
If this is run as es2015 then it'll be slower due to the use of `for.. of`

## What is the new behavior?
Same performance but without relying on the final target being es5 with `--downlevelIteration` 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by alexeagle — APPROVED** [review]

**Comment by alfaproject** [comment]
It used to be slower, but isn't it almost the same nowadays? Is it really worth degrading the readability in these cases?

**Comment by Toxicable** [comment]
@alfaproject No this is still a well known issue.
For a hot path like any part of a compiler I think the readability is worth the trade off, no one wants a slow compiler

File: packages/bazel/src/ngc-wrapped/tsconfig.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
   "compilerOptions": {
-    "lib": ["es5", "es2015.collection", "es2015.core"]
+    "lib": ["es5", "es2015.collection", "es2015.core", "es2015.iterable"]
```
Comment (IgorMinar) [inline packages/bazel/src/ngc-wrapped/tsconfig.json]: do we need this change? it doesn't seem that way to me.

**Review by IgorMinar — CHANGES_REQUESTED** [review]

File: packages/bazel/src/ngc-wrapped/tsconfig.json (hunk 1 above)
Comment (Toxicable) [reply packages/bazel/src/ngc-wrapped/tsconfig.json]: It's needed for rest and spread parameters.
Such as here:  https://github.com/Toxicable/angular/blob/62e01b789ed5bd2b07562d7451578b68afed2a30/packages/bazel/src/ngc-wrapped/index.ts#L359

Do you want me to refactor those out ontop of it?

Comment (Toxicable) [reply packages/bazel/src/ngc-wrapped/tsconfig.json]: Turns out this flag isn't required for rest and spread so i've removed the flag

**Review by IgorMinar — APPROVED** [review]

**Pull request closed without merging**